Mathlib/GroupTheory/GroupAction/Defs.lean

```diff
@@ -0,0 +1,1194 @@
+/-
+Copyright (c) 2018 Chris Hughes. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Chris Hughes, Yury Kudryashov
+-/
+import Mathlib.Algebra.Group.TypeTags
+import Mathlib.Algebra.Group.Commute
+import Mathlib.Algebra.Hom.Group
+import Mathlib.Algebra.Opposites
+import Mathlib.Logic.Embedding.Basic
+
+/-!
+# Definitions of group actions
+
+This file defines a hierarchy of group action type-classes on top of the previously defined
+notation classes `SMul` and its additive version `VAdd`:
+
+* `MulAction M α` and its additive version `AddAction G P` are typeclasses used for
+  actions of multiplicative and additive monoids and groups; they extend notation classes
+  `SMul` and `VAdd` that are defined in `Algebra.Group.Defs`;
+* `DistribMulAction M A` is a typeclass for an action of a multiplicative monoid on
+  an additive monoid such that `a • (b + c) = a • b + a • c` and `a • 0 = 0`.
+
+The hierarchy is extended further by `module`, defined elsewhere.
+
+Also provided are typeclasses for faithful and transitive actions, and typeclasses regarding the
+interaction of different group actions,
+
+* `SMulCommClass M N α` and its additive version `VAddCommClass M N α`;
+* `IsScalarTower M N α` and its additive version `VAddAssocClass M N α`;
+* `IsCentralScalar M α` and its additive version `IsCentralVAdd M N α`.
+
+## Notation
+
+- `a • b` is used as notation for `SMul.smul a b`.
+- `a +ᵥ b` is used as notation for `VAdd.vadd a b`.
+
+## Implementation details
+
+This file should avoid depending on other parts of `GroupTheory`, to avoid import cycles.
+More sophisticated lemmas belong in `GroupTheory.GroupAction`.
+
+## Tags
+
+group action
+-/
+
+
+variable {M N G A B α β γ δ : Type _}
+
+open Function (Injective Surjective)
+
+/-!
+### Faithful actions
+-/
+
+
+/-- Typeclass for faithful actions. -/
+class HasFaithfulVadd (G : Type _) (P : Type _) [VAdd G P] : Prop where
```

Comment:
I think our capitalisation rule is that this should be `HasFaithfulVAdd`, and similarly for `SMul`.